aten/src/ATen/cpu/vec/vec512/vec512_int.h

```diff
@@ -774,58 +778,72 @@ class Vectorized<int8_t> : public Vectorizedi {
       // https://software.intel.com/content/www/us/en/develop/documentation/cpp-compiler-developer-guide-and-reference/top/compiler-reference/intrinsics/intrinsics-for-intel-advanced-vector-extensions/intrinsics-for-load-and-store-operations-1/mm512-storeu-si512.html
       _mm512_storeu_si512(reinterpret_cast<__m512i*>(ptr), values);
     } else if (count > 0) {
-      __at_align__ int8_t tmp_values[size()];
+      __at_align__ T tmp_values[size()];
       _mm512_storeu_si512(reinterpret_cast<__m512i*>(tmp_values), values);
-      std::memcpy(ptr, tmp_values, count * sizeof(int8_t));
+      std::memcpy(ptr, tmp_values, count * sizeof(T));
     }
   }
-  const int8_t& operator[](int idx) const  = delete;
-  int8_t& operator[](int idx)  = delete;
-  Vectorized<int8_t> abs() const {
-    return _mm512_abs_epi8(values);
+  const T& operator[](int idx) const  = delete;
+  T& operator[](int idx)  = delete;
+  Vectorized<T> abs() const {
+    if (std::is_same<T, int8_t>::value)
+      return _mm512_abs_epi8(values);
+    else
+      return values;
   }
-  Vectorized<int8_t> real() const {
+  Vectorized<T> real() const {
     return *this;
   }
-  Vectorized<int8_t> imag() const {
+  Vectorized<T> imag() const {
     return _mm512_set1_epi8(0);
   }
-  Vectorized<int8_t> conj() const {
+  Vectorized<T> conj() const {
     return *this;
   }
-  Vectorized<int8_t> frac() const;
-  Vectorized<int8_t> neg() const;
-  Vectorized<int8_t> operator==(const Vectorized<int8_t>& other) const {
-    auto mask = _mm512_cmpeq_epi8_mask(values, other.values);
+  Vectorized<T> frac() const;
+  Vectorized<T> neg() const;
+  Vectorized<T> operator==(const Vectorized<T>& other) const {
+    __mmask64 mask;
+    if (std::is_same<T, int8_t>::value)
+      mask = _mm512_cmpeq_epi8_mask(values, other.values);
+    else
+      mask = _mm512_cmpeq_epu8_mask(values, other.values);
     return _mm512_mask_set1_epi8(zero_vector, mask, 0xFF);
   }
-  Vectorized<int8_t> operator!=(const Vectorized<int8_t>& other) const {
-    auto mask = _mm512_cmpneq_epi8_mask(values, other.values);
+  Vectorized<T> operator!=(const Vectorized<T>& other) const {
+    if (std::is_same<T, int8_t>::value)
+      mask = _mm512_cmpneq_epi8_mask(values, other.values);
```

Comment:
Build is failing because `mask` has not been declared in this function & the two below it.

Reply:
Thanks - fixed, and it passes CI now.